bench/rttanalysis: TestBenchmarkExpectationShard3 failed

bench/rttanalysis.TestBenchmarkExpectationShard3 [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21293100?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21293100?buildTab=artifacts#/) on master @ [cc1040617836359ffc8d6c16f95914f14b6461cc](https://github.com/cockroachdb/cockroach/commits/cc1040617836359ffc8d6c16f95914f14b6461cc):


```
=== RUN   TestBenchmarkExpectationShard3
    validate_benchmark_data.go:186: [/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18902/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/bench/rttanalysis/rttanalysis_test_/rttanalysis_test.runfiles/com_github_cockroachdb_cockroach/pkg/bench/rttanalysis/rttanalysis_test_/rttanalysis_test --test.v=test2json --test.bench=^$]
I260401 10:53:40.133991 1 (gostd) rand.go:288  [-] 1  random seed: -7518734791283483324
=== RUN   TestBenchmarkExpectationShard3
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/0a9e3bf541e53bd0aafa6620442dfec3/logTestBenchmarkExpectationShard3107253287
    test_log_scope.go:82: use -show-logs to present logs inline
    validate_benchmark_data.go:120: success: expected GenerateObjects/generate_1000_tables_-_this_test_should_use_the_same_number_of_RTTs_as_for_10_tables to perform KV lookups in [6, 6], got 5
    validate_benchmark_data.go:120: success: expected GenerateObjects/generate_10_tables to perform KV lookups in [6, 6], got 5
    validate_benchmark_data.go:120: success: expected GenerateObjects/generate_50000_tables to perform KV lookups in [6, 6], got 5
    validate_benchmark_data.go:120: success: expected AlterTableSplit/alter_table_split_at_1_value to perform KV lookups in [8, 8], got 8
    validate_benchmark_data.go:120: success: expected GenerateObjects/generate_100_tables_from_template to perform KV lookups in [9, 9], got 8
    validate_benchmark_data.go:120: success: expected AlterTableSplit/alter_table_split_at_2_values to perform KV lookups in [11, 11], got 11
    validate_benchmark_data.go:120: success: expected AlterTableSplit/alter_table_split_at_3_values to perform KV lookups in [14, 14], got 14
    validate_benchmark_data.go:120: success: expected Revoke/revoke_all_on_2_tables to perform KV lookups in [8, 8], got 8
    validate_benchmark_data.go:120: success: expected Revoke/revoke_all_on_1_table to perform KV lookups in [8, 8], got 8
    validate_benchmark_data.go:120: success: expected DropDatabase/drop_database_2_tables to perform KV lookups in [16, 16], got 16
    validate_benchmark_data.go:120: success: expected DropDatabase/drop_database_1_table to perform KV lookups in [16, 16], got 16
    validate_benchmark_data.go:120: success: expected Truncate/truncate_2_column_2_rows to perform KV lookups in [17, 17], got 17
    validate_benchmark_data.go:120: success: expected AlterTableAddColumn/alter_table_add_1_column to perform KV lookups in [13, 13], got 13
    validate_benchmark_data.go:120: success: expected DropDatabase/drop_database_3_tables to perform KV lookups in [16, 16], got 16
    validate_benchmark_data.go:120: success: expected Revoke/revoke_all_on_3_tables to perform KV lookups in [8, 8], got 8
    validate_benchmark_data.go:120: success: expected AlterTableAddColumn/alter_table_add_2_columns to perform KV lookups in [13, 13], got 13
    validate_benchmark_data.go:120: success: expected Truncate/truncate_2_column_1_rows to perform KV lookups in [17, 17], got 17
    validate_benchmark_data.go:120: success: expected Truncate/truncate_2_column_0_rows to perform KV lookups in [18, 18], got 18
    validate_benchmark_data.go:120: success: expected Truncate/truncate_1_column_2_rows to perform KV lookups in [17, 17], got 17
    validate_benchmark_data.go:120: success: expected AlterTableAddColumn/alter_table_add_3_columns to perform KV lookups in [13, 13], got 13
    validate_benchmark_data.go:120: success: expected Truncate/truncate_1_column_1_row to perform KV lookups in [17, 17], got 17
    validate_benchmark_data.go:120: success: expected Truncate/truncate_1_column_0_rows to perform KV lookups in [18, 18], got 18
    validate_benchmark_data.go:117: fail: expected DropDatabase/drop_database_0_tables to perform KV lookups in [15, 15], got 17
    validate_benchmark_data.go:120: success: expected GenerateObjects/generate_10x10_schemas_and_tables_in_existing_db to perform KV lookups in [17, 17], got 16
    validate_benchmark_data.go:73: see the -rewrite flag to re-run the benchmarks and adjust the expectations
    validate_benchmark_data.go:74: usage: ./dev test --rewrite pkg/bench/rttanalysis -f=TestBenchmarkExpectation/ --test-args '-rewrite-iterations=N'
    validate_benchmark_data.go:107: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/0a9e3bf541e53bd0aafa6620442dfec3/logTestBenchmarkExpectationShard3107253287
--- FAIL: TestBenchmarkExpectationShard3 (213.85s)
--- FAIL: TestBenchmarkExpectationShard3 (214.86s)
=== RUN   TestBenchmarkExpectationShard3/DropDatabase
--- FAIL: TestBenchmarkExpectationShard3/DropDatabase (190.17s)
=== RUN   TestBenchmarkExpectationShard3/DropDatabase/drop_database_0_tables
    rtt_analysis_bench.go:224: DropDatabase/drop_database_0_tables: got 17, expected 15
    rtt_analysis_bench.go:230: wrote jaeger trace to /artifacts/tmp/_tmp/0a9e3bf541e53bd0aafa6620442dfec3/logTestBenchmarkExpectationShard3107253287/DropDatabase_drop_database_0_tables.jaeger.json
--- FAIL: TestBenchmarkExpectationShard3/DropDatabase/drop_database_0_tables (190.17s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBenchmarkExpectationShard3.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-62358

--- a/pkg/bench/rttanalysis/testdata/benchmark_expectations
+++ b/pkg/bench/rttanalysis/testdata/benchmark_expectations
@@ -53,11 +53,11 @@ exp,benchmark
 16,CreateRole/create_role_with_2_options
 19,CreateRole/create_role_with_3_options
 14,CreateRole/create_role_with_no_options
-15,DropDatabase/drop_database_0_tables
+15-17,DropDatabase/drop_database_0_tables
 16,DropDatabase/drop_database_1_table
 16,DropDatabase/drop_database_2_tables
 16,DropDatabase/drop_database_3_tables
-19-21,DropRole/drop_1_role
+19-22,DropRole/drop_1_role
 28-30,DropRole/drop_2_roles
 37-39,DropRole/drop_3_roles
 13,DropSequence/drop_1_sequence

